### ISSUE TITLE
Softmax + cross_entropy_cost报错： Check failed: dynamic_cast<CpuIVector*>(&label)

### ISSUE THREAD

**Issue by dockiHan** [PaddlePaddle/Paddle#3716]
最近使用paddlepaddle搭建一个简单的二分类分类器，输出层激活函数使用Softmax，损失函数使用cross_entropy_cost，单机CPU运行时报如下错误：
![image](https://user-images.githubusercontent.com/22857939/29758291-d31c2ab4-8be4-11e7-8fb5-3dfdeca3620c.png)

由于没有更加详细的错误提示，实在找不出错误在哪里，还望各位老师多加指点。以下是网络搭建的详细代码：
```python
#Define network
X = paddle.layer.data(name="X", type = paddle.data_type.dense_vector(dim=input_dim))
Y = paddle.layer.data(name="Y", type = paddle.data_type.dense_vector(dim=output_dim))

def logistic_regression(input, hide_act, hidden_dim, out_act, output_dim):
    hide_lr = 1e-3
    out_lr = 1e-3
    hide_param_attr = paddle.attr.Param(learning_rate=hide_lr, gradient_clipping_threshold=100)
    out_param_attr = paddle.attr.Param(learning_rate=out_lr, gradient_clipping_threshold=100)
    hide_0 = paddle.layer.fc(input=input, size=hidden_dim, act=hide_act, param_attr=hide_param_attr)
    out_layer = paddle.layer.fc(input=hide_0, size=output_dim, act=out_act, param_attr=out_param_attr)
    return out_layer

output = logistic_regression(input=X, 
                             hide_act=paddle.activation.Relu(), 
                             hidden_dim=hidden_dim, 
                             out_act=paddle.activation.Softmax(), 
                             output_dim=output_dim)

cost = paddle.layer.cross_entropy_cost(input=output, label=Y)
parameters = paddle.parameters.create(cost)
optimizer = paddle.optimizer.Adam()
trainer = paddle.trainer.SGD(cost=cost, parameters=parameters, update_equation=optimizer)
```

**Comment by wanghaoshuang** [comment]
Please change
```
Y = paddle.layer.data(name="Y", type = paddle.data_type.dense_vector(dim=output_dim))
```
to 
```
Y = paddle.layer.data(name="Y", type = paddle.data_type.integer_value(dim=output_dim))
```

**Comment by dockiHan** [comment]
@wanghaoshuang
看API中paddle.data_type.integer_value 类型并没有参数dim， 请问integer_value 中的value_range跟dim效果是一样的吗？

**Comment by wanghaoshuang** [comment]
@dockiHan 不一样，他们分别是：
:param dim: dimension of this vector.
:param value_range: range of this integer.

**Comment by dockiHan** [comment]
@wanghaoshuang  恩恩，明白了，谢谢！
